### ISSUE TITLE
'MODULE_NOT_FOUND' when converting pdf to epub [ electron with CRA ]

### ISSUE THREAD

**Issue by 61soldiers** [JVKdouk/node-ebook-converter#1]
In main process

```js
let pdfPath = path.parse(data.path)
// Convert with calibre
ebookConverter.convert({
  input: data.path,
  output: `${pdfPath.dir}/${pdfPath.name}.epub`,
  silent: true,
}).then(async (res) => await event.reply('convertPdfToEpub', {status: 'success'}))
  .catch(async (err) => await event.reply('convertPdfToEpub', {status: 'failed', err: err}))
```

It works in development but once i build the electron app and try to convert that error shows up.
Note: I also have a check to see if `ebook-convert` command can be executed so the conversion only runs if check returns true

**Comment by JVKdouk** [comment]
I will be checking what is going on. Does the error mentioned follows a Stack trace? If so, can you please paste it here?

**Comment by 61soldiers** [comment]
No, i passed in the error object as you can see from the code up there, it logs this

![image](https://user-images.githubusercontent.com/34519591/104128500-b196b100-5389-11eb-827c-01c1617eb96a.png)

It just outputs `MODULE_NOT_FOUND`

**Comment by JVKdouk** [comment]
Reinstalling Node fixes your problem? If not, trying reinstalling your node_modules please!

**Comment by JVKdouk** [comment]
https://stackoverflow.com/questions/37994441/how-to-use-fs-module-inside-electron-atom-webpack-application

I found this. It may be helpful for Electron apps. I won't be able to check it right now, I believe I will try this solution over the week.

**Comment by 61soldiers** [comment]
Don't think that will be helpful as i'm converting in the main process, anyways thanks for looking into it. I reinstalled ( Updated ) to the latest node 12 version ( as node 14 has breaking changes ) and reinstalled node_modules as you said, didn't work. Same error.
Could be a problem with webpack but i'm not sure since the file is being converted in the backend ( main process ).

**Comment by JVKdouk** [comment]
So, just dug a bit here and the problem is a little deeper than a initially thought. Node workers doesn't work natively in electron apps and the path system breaks with the latest version of it. I will need some more time to make a fix for it a probably use background workers instead for Chromium.
